### PR TITLE
Update Auth Service and Adapt TOTP Endpoints

### PR DESCRIPTION
**Pull request by dontseyit**

### CODE REVIEW

**Comment by dontseyit** [comment]
The dependency conflict is postponed to another PR. Also the comment `# user_info = {..}` is forgotten and has no effect, but I don't want to wait another 20 minutes for tests just for this, so it'll be removed later as well.

File: fixtures/auth.py [hunk 1]
```diff
@@ -277,6 +277,17 @@ def auth_login(self, headers: dict[str, Any]):
...
+    def add_totp_to_headers(
+        self, totp_token: str, headers: dict[str, Any]
+    ) -> dict[str, Any]:
+        """Add TOTP token to headers.
+
+        'X-Authorization' header is used to submit the one-time password. Due to ExtAuth
+        protocol by default doesn't allow the request body.
+        """
+        headers["X-Authorization"] = f"Bearer TOTP:{totp_token}"
+        return headers
```
Comment (Cito) [inline fixtures/auth.py:289]: Usually, functions that modify objects in place return None in Python.

File: fixtures/auth.py [hunk 2]
```diff
@@ -321,9 +331,10 @@ def generate_totp(
 
     def verify_totp(self, user_id: str, totp: str, headers: dict[str, Any]) -> Response:
         """Verify the TOTP code."""
-        user_info = {"user_id": user_id, "totp": totp}
+        # user_info = {"user_id": user_id, "totp": totp}
         url = self.auth_adapter_url + "/rpc/verify-totp"
-        return self.http.post(url, json=user_info, headers=headers)
+        headers = self.add_totp_to_headers(totp, headers)
```
Comment (Cito) [inline fixtures/auth.py:336]:
```suggestion
        self.add_totp_to_headers(totp, headers)
```

**Review by Cito — APPROVED** [review]
Looks good to me, just one side note.

File: fixtures/auth.py (hunk 1 above)
Comment (Cito) [reply fixtures/auth.py:289]: Would also change the order of the parameters to make that clearer.

Comment (dontseyit) [reply fixtures/auth.py:289]: Thank you for drawing attention to this, I'll fix it in the next PR.

**Pull request merged**